### PR TITLE
adding custom route handling, Resolves #17

### LINKED ISSUE #17: header checking not working
* **I'm submitting a ...**
  - [x] bug report
  -  feature request
  -  support request

* **Do you want to request a *feature* or report a *bug*?**

Bug with Header authorization checking.

* **What is the current behavior?**
Currently regardless of the value of the "Authorization" header the user is allowed to interact with any of the endpoints and send and receive data from them.

* **If the current behavior is a bug, please provide the steps to reproduce the behavior.**
Any properly formed POST to one of the endpoints will return the expected value from that endpoint. 

* **What is the expected behavior?**

If the value of the "Authorization" header is `DS_SECRET_TOKEN` the the application works as expected and returns the appropriate data from the endpoints.

if the value of Authorization header is not `DS_SECRET_TOKEN` then a 403: forbidden error is returned to the user instead of the expected data from the endpoint call.

* **What is the motivation / use case for changing the behavior?**

Informational Security and compliance with COPPA regulations

* **Please tell us about your environment:**
  
  Operating system and current build version: Win, WSL2: `4.4.0-19041-Microsoft #488-Microsoft Mon Sep 01 13:43:00 PST 2020 x86_64 x86_64 x86_64 GNU/Linux`
  
  Output from `pip freeze` for the following packages:
 
  on `python 3.7.6`
  ```
  fastapi==0.60.1
  pandas
  numpy
  plotly
  uvicorn==0.11.8
  google-cloud-vision==1.0.0
  textstat==0.6.2
  ```
  
  output from command:
 
  `python 3.7.6`
  `python 3.8.2`

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

### PR DESCRIPTION
**Pull request by Jwilson1172**
Close #17 

^^^
keyword for GitHub to link the issue to this PR

## Description

Fixes #17 by adding a custom APIRoute class passed to the APIRouter in submission.py to enforce `DS_SECRET_TOKEN` checking.
During testing I found that `ValidationError` from the data models was being raised to `AuthRouteHandler` then not being caught so I added a catch statement to handle that Exception from pydantic

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## Change Status

- [X] Complete, tested, ready to review and merge
- Complete, but not tested (may need new tests)
- Incomplete/work-in-progress, PR is for discussion/feedback

## Has This Been Tested

- [X] Yes
- No
- Not necessary

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- [X] There are no merge conflicts

### CODE REVIEW

File: project/app/utils/security/header_checking.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from fastapi.responses import JSONResponse
+from fastapi.routing import APIRoute
+from pydantic import ValidationError
+
+
+class AuthRouteHandler(APIRoute):
+    """Custom APIRoute handler that checks the authorization header for
+    secret value
+    more information about this method for overriding the APIRouter handling can be found here:
+    https://fastapi.tiangolo.com/zh/advanced/custom-request-and-route/
```
Comment (lorischl-otter) [inline project/app/utils/security/header_checking.py:14]: Good call on including the link for more info

File: project/app/utils/security/header_checking.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            # unexpected error
+            except Exception as e:
+                print(
+                    "error at app.utils.security.header_checking.AuthRouteHandler()",
+                    e.with_traceback(None))
+                return JSONResponse(
+                    status_code=500,
+                    content={"ERROR": "Unknown error please contact sysadmin"})
+            finally:
+                # remove secret from mem after using it to authenicate.
```
Comment (schase15) [inline project/app/utils/security/header_checking.py]: *memory, *authenticate

**Review by schase15 — CHANGES_REQUESTED** [review]
Looks great! Thank you for fixing that. Just caught two typos down at the bottom

**Review by lorischl-otter — DISMISSED** [review]
Glad you got this figured out! Thanks for getting the fix implemented.

**Review by schase15 — APPROVED** [review]
Thank you!

**Review by lorischl-otter — APPROVED** [review]

**Pull request merged**